README.md

```diff
@@ -984,6 +984,10 @@ in inheritance.
   is a hybrid of `Array`'s intuitive inter-operation facilities and
   `Hash`'s fast lookup.
 
+*  Prefer `map` over `collect`, `detect` over `find`, and `reduce` over
```

Comment:
detect over find is outlined in the actual pr. Maybe @tommydangerous can outline the preference for map, inject, etc.